### PR TITLE
Bug/en 3214 fix bug in "Has" method from nonceSyncMapCacher

### PR DESCRIPTION
**Pull request by SebastianMarian**
* Fixed "Has" method from nonceSyncMapCacher
* ErrInvalidBlockHash has been split in two different errors (one for current shard blocks and one for notarized blocks) to help debugging

### CODE REVIEW

File: dataRetriever/dataPool/nonceSyncMapCacher.go [hunk 1]
```diff
@@ -143,9 +147,19 @@ func (nspc *nonceSyncMapCacher) callAddedDataHandlers(nonce uint64, shardId uint
 	nspc.mutAddedDataHandlers.RUnlock()
 }
 
-// Has returns true if a map is found for provided nonce
-func (nspc *nonceSyncMapCacher) Has(nonce uint64) bool {
-	_, ok := nspc.cacher.Peek(nspc.nonceConverter.ToByteSlice(nonce))
+// Has returns true if a map is found for provided nonce ans shardId
+func (nspc *nonceSyncMapCacher) Has(nonce uint64, shardId uint32) bool {
+	val, ok := nspc.cacher.Peek(nspc.nonceConverter.ToByteSlice(nonce))
```
Comment (raduchis) [inline dataRetriever/dataPool/nonceSyncMapCacher.go]: Just an observation: Has does everything like the Get method. If there are calls to Has and exactly after Get, use only get to not load everything twice.

File: process/block/baseProcess.go [hunk 2]
```diff
@@ -211,7 +211,7 @@ func (bp *baseProcessor) isHdrConstructionValid(currHdr, prevHdr data.HeaderHand
 	}
 
 	if !bytes.Equal(currHdr.GetPrevHash(), prevHeaderHash) {
-		return process.ErrInvalidBlockHash
+		return process.ErrInvalidNotarizedBlockHash
```
Comment (raduchis) [inline process/block/baseProcess.go]: ErrInvalidNotarizedBlockHash? Is it invalid, or they do not match?

Comment (SebastianMarian) [reply process/block/baseProcess.go]: Actually it could be ErrBlockHashDoesNotMatch

File: dataRetriever/dataPool/nonceSyncMapCacher.go (hunk 1 above)
Comment (SebastianMarian) [reply dataRetriever/dataPool/nonceSyncMapCacher.go]: I verified this and it is used only in one single place where I also found this bug.

File: process/block/baseProcess.go (hunk 2 above)
Comment (SebastianMarian) [reply process/block/baseProcess.go]: Renamed

File: dataRetriever/dataPool/nonceSyncMapCacher.go (hunk 1 above)
Comment (raduchis) [reply dataRetriever/dataPool/nonceSyncMapCacher.go]: ok

File: process/block/baseProcess.go (hunk 2 above)
Comment (raduchis) [reply process/block/baseProcess.go]: ok

File: integrationTests/consensus/consensus_test.go [hunk 3]
```diff
@@ -49,7 +49,7 @@ func initNodesAndTest(numNodes, consensusSize, numInvalid uint32, roundTime uint
 		for i := uint32(0); i < numInvalid; i++ {
 			nodes[i].blkProcessor.ProcessBlockCalled = func(blockChain data.ChainHandler, header data.HeaderHandler, body data.BodyHandler, haveTime func() time.Duration) error {
 				fmt.Println("process block invalid ", header.GetRound(), header.GetNonce(), getPkEncoded(nodes[i].pk))
```
Comment (AdoAdoAdo) [inline integrationTests/consensus/consensus_test.go]: should we also change the logged message to match the returned error?

Comment (SebastianMarian) [reply integrationTests/consensus/consensus_test.go]: I think that this print is something generic here and anyway the returned error will be printed in the caller side to spot the exactly error occurred

**Pull request merged**